library/gensym.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/python
+
+import subprocess
+from ansible.module_utils.basic import AnsibleModule 
+
+def test_name(name, used_list):
+    for item in used_list:
+        if name in item:
+            return False
+    return True
+
+def init_volume(result, module):
+    # Check if fs_type is 'swap', use that instead of mount pt
+    if 'swap' in module.params['fs_type'].lower():
+        result['volume_name'] = 'swap'
+
+    # Check if the mount point of fs is root 
+    else:
+        if module.params['mount'] == '/':
+            result['volume_name'] = 'root'
+        else:
+            result['volume_name'] = module.params['mount'][1:].replace('/', '_')
+
+    def get_lvcmd_buf():
+        lvs_cmd = subprocess.check_output('sudo lvs --no-headings -o name', shell = True)
+        buf = filter(None, lvs_cmd.lower().replace(' ', '').split('\n'))
+        return buf 
+
+    buf = get_lvcmd_buf()
+    counter = 0
+        
+    unique_name = test_name(result['volume_name'], buf)    
+    while not unique_name:
+        if counter == 0:
+            result['volume_name'] = result['volume_name'] + str(counter)
+        else:
+            counter += 1
+            result['volume_name'] = result['volume_name'][:-1] + str(counter)
+        unique_name = test_name(result['volume_name'], buf)
+
+def init_group(result):
+    # Get the command buffer, parse into a list 
+    vgs_cmd = subprocess.check_output("sudo vgs --no-headings -o name", shell = True)
+    buf = vgs_cmd.strip().split('\n') 
+
+    def get_os():    
+        for line in open('/etc/os-release').readlines():
+            if not line.find('ID='):
+                name = line[3:]
+            
+        name = name.replace('\n', '').replace('"', '')
+        return name 
+
+    def get_host():
+        host_buf = subprocess.check_output('hostname', shell = True)
+        buf = host_buf.replace('\n', '').replace('.', '_').replace('-', '_')
+        return buf
+
+    # Define necessary variables
+    os_name = get_os()
+    host_name = get_host()
+    default_name = get_os() + '_' + host_name
+    result['group_name'] = default_name
+    unique_name = test_name(result['group_name'], buf) 
+    counter = 0
+
+    # Test generated name is unique
+    while not unique_name:
+        if counter == 0:
+            result['group_name'] = result['group_name'] + str(counter)
+        else:
+            counter += 1
+            result['group_name'] = result['group_name'][:-1] + str(counter)
+        unique_name = test_name(result['group_name'], buf)
+
+def run_module():
+    module_args = dict(
+        mount=dict(type='str', required=True),
+        fs_type=dict(type='str', required=True)
+    )
+
+    result = dict(
+        changed=True,
+        group_name='',
```

Comment:
For consistency, it would be good if the result dict keys were `vg_name` and `lv_name`.